kernel/safe_typing.ml

```diff
@@ -1194,39 +1194,46 @@ let register_inline kn senv =
   let cb = {cb with const_inline_code = true} in
   let env = add_constant kn cb env in { senv with env}
 
-let check_register_ind (mind,i) r env =
-  let mb = Environ.lookup_mind mind env in
-  let check_if b s =
+let check_register_ind ind r env =
+  let (mb,ob as spec) = Inductive.lookup_mind_specif env ind in
```

Comment:
If we do care you can ask that `List.mem Sorts.(family_eq InType) ob.mind_kelim` ie the inductive is not squashed.

BTW usually we call the one_inductive_body "mip" (p for packet since it comes from mind_packets) not "ob"